_posts/2021-07-05-newsletter.md

```diff
@@ -0,0 +1,199 @@
+
+For a comprehensive and up-to-date guide on using Lua for Neovim scripting and configuration, see  [Getting started using Lua in Neovim](https://github.com/nanotee/nvim-lua-guide). A good model for a Lua configuration using `init.lua` is [Defaults.nvim](https://github.com/mjlbach/defaults.nvim).
+
+
+## Language Server Protocol (LSP)
+
+The [Language Server Protocol](https://microsoft.github.io/language-server-protocol/) (LSP) is an open, JSON-RPC-based protocol for communication between code editors and language servers, which provide programming language-specific features such as
+
+- completion,
+- hover/tooltips,
+- go to definition,
+- show/go to references,
+- show method signatures,
+- rename,
+- code actions (automatic formatting, organize imports, ...),
+
+and more.
+
+The idea is to separate these features into an editor-independent but language-specific server and a language-independent but editor-specific client, which communicate via the Language Server Protocol over RPC. (It should be pointed out that not every server implements every feature, and the quality of the responses can vary wildly. The "reference implementation" in VS Code also often adds non-standard features that are not covered by the LSP itself).
+
+Neovim 0.5 provides an LSP client written (mostly) in Lua that provides a highly configurable and extensible way of accessing these features. It does not aim at competing with more feature-rich and "out-of-the-box" plugins like [CoC.nvim](https://github.com/neoclide/coc.nvim) but is meant to be tailored to your preferences (while still being useable with reasonable defaults). For an overview, see TJ DeVries' [Vimconf.live presentation](https://www.youtube.com/watch?v=C9X5VF9ASac) and his [shorter video](https://www.youtube.com/watch?v=ArwDgvYEZYk).
+
+For many language servers, [Nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) already provides the necessary configuration to set everything up easily. Alternatively, some languages also have specific LSP plugins that provide a more integrated setup, e.g., for [Java](https://github.com/mfussenegger/nvim-jdtls) and [Scala](https://github.com/ckipp01/nvim-metals).
+
+To learn more about LSP and how to use it in Neovim, visit [Nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) and read [`:h lsp`](https://neovim.io/doc/user/lsp.html).
+
+Expect more work on LSP to provide more configuration options and full coverage of LSP version 3.16 (including semantic highlighting).
```

Comment:
`Expect better coverage of the LSP spec (including, semantic highlighting) in future releases`

Reply:
I was referring to `lsp.config` here.

Reply:
And I think we are pretty good on 3.15 coverage; it's just the newest version (that got published during development) that has gaps?